File: CodeforcesContests/div1_pacific_northwest18-19_icpc/J.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void){
    int n, s, input;
    cin >> n >> s;
    int max_time = 0;
    for(int i = 0; i < n; ++i){
        cin >> input;
        max_time = max(max_time, input);
    }
    max_time = max_time * s;
    cout << (max_time+1000-1)/1000 << endl; // Gives the ceil of the division max_time/1000.
    return 0;
}

File: simulaciones/11_06_2023/J.py





def main():
    n = int(input())
    s = input()

    ans = 1
    i = 1
    while i < n:
        if 2*i <= n and s[0:i] + s[0:i] == s[0:2*i]:
            ans += 1
            i *= 2
        else:
            ans += 1
            i += 1

    print(ans)

main()


